specs/experimental/inbox-contract.md

```diff
@@ -55,179 +58,126 @@ These modifications aim to enhance the security and efficiency of the batch subm
+/// @param _value The encoded value with which to set the batch inbox address.
+function _setBatchInbox(bytes calldata _value) internal {
+    batchInbox = StaticConfig.decodeSetBatchInbox(_value);
 }
 ```
 
-### L1Block
+### StaticConfig
 
-The `L1Block` stores Layer 1 (L1) related information on Layer 2 (L2). It is extended to store the dynamic inbox address.
+The `StaticConfig` library is used for encoding and decoding static configuration data.
 
-#### batchInbox
+#### encodeSetBatchInbox
 
-A new field `batchInbox` is added to the `L1Block`:
+This function is used on L1 to encode the static configuration data for setting a batch inbox.
 
 ```solidity
-    /// @notice The canonical batch inbox.
-    bytes32 public batchInbox;
+/// @notice Encodes the static configuration data for setting a batch inbox.
+/// @param _batchInbox Address of the batch inbox.
+/// @return Encoded static configuration data.
+function encodeSetBatchInbox(address _batchInbox) internal pure returns (bytes memory) {
+    return abi.encode(_batchInbox);
+}
 ```
 
-#### setL1BlockValuesEcotone
+#### decodeSetBatchInbox
 
-This function stores Layer 1 (L1) block values for Layer 2 (L2) since the Ecotone upgrade of the OP Stack. It is enhanced to also store the inbox address.
+This function is used on L2 to decode the static configuration data for setting a batch inbox.
 
 ```solidity
-function setL1BlockValuesEcotone() external {
-    ...
-    sstore(batchInbox.slot, calldataload(164)) // bytes32
+/// @notice Decodes the static configuration data for setting a batch inbox.
+/// @param _data Encoded static configuration data.
+/// @return Decoded Address of the batch inbox.
+function decodeSetBatchInbox(bytes memory _data) internal pure returns (address) {
+    return abi.decode(_data, (address));
 }
-
 ```
 
+### L2Client.SystemConfigByL2Hash
+
+This function is used to retrieve the system configuration for a specific L2 block hash. Prior to the fork, this information was obtained from the deposit transaction. However, post-fork, it should be read from the L2 state instead, in accordance with the concepts outlined in [this issue](https://github.com/ethereum-optimism/specs/issues/122).
+
+### L1Traversal.AdvanceL1Block
+
+This function is used to fetch the next L1 block. Its behavior differs pre- and post-fork:
+
+Pre-fork:
+- It internally caches the post-state of the system configuration for the current L1 block.
+- This cached configuration is then utilized by `DataAvailabilitySource.OpenData` to derive batches, as seen [here](https://github.com/ethereum-optimism/optimism/blob/0cf2a11611d329b9c6d53fd6473dbbc44f68e94c/op-node/rollup/derive/l1_retrieval.go#L57).
+
+Post-fork:
+- It consistently reads the system configuration from the L2 state of the parent block.
+
+
+
 ### How `op-node` knows the canonical batch inbox
 
 We define that the canonical batch inbox at a specific L2 block is the batch inbox of `SystemConfig` of the origin of the L2 block.
```

Comment:
Should this part be updated accordingly?

Reply:
Updated.

Reply:
Updated.